### ISSUE TITLE
Updating components such as pubsub, the old components is never closed when running in a Kubernetes mode.

### ISSUE THREAD

**Issue by 63hogan** [dapr/dapr#4218]
## In what area(s)?
pkg/runtime/runtime.go

## What version of Dapr?
newest

## Expected Behavior
close old component before updating component with same pubsubname.

## Actual Behavior
when updating some pubsub component config,  the old should be closed.
```go
// pkg/runtime/runtime.go
func (a *DaprRuntime) onComponentUpdated(component components_v1alpha1.Component) {
}

func (a *DaprRuntime) initPubSub(c components_v1alpha1.Component) error {
a.pubSubs[pubsubName] = pubSub
}
```
when received update request, if the pubsubName is same as old component(maybe for  updating config?) it just replaced with the new component pusSub, but the old is never closed.

**Comment by yaron2** [comment]
Thanks, will be fixed in 1.7.

**Comment by mstrYoda** [comment]
> 

@yaron2 There is also another issue which I opened a time ago, it is automatically closed, is that issue will be fixed in 1.7 too? https://github.com/dapr/dapr/issues/3015

**Comment by shubham1172** [comment]
FYI, to reproduce the issue:
- Deploy the tutorial hello-world: https://github.com/dapr/quickstarts/tree/master/tutorials/hello-kubernetes
- Make changes to the state store configuration and run `kubectl apply -f redis-state.yaml`
- With each run, there is an extra Redis client added (can be seen in the screenshot) 

![image](https://user-images.githubusercontent.com/22132944/159647057-948c3e20-b3e6-46e3-8e57-4e1c2007c366.png)

**Comment by shubham1172** [comment]
After the PR: 
(1 is CLI, 1 is GUI, 1 is Dapr)
![image](https://user-images.githubusercontent.com/22132944/159654703-6d2ea00d-2982-4bd5-bd70-38dea1275120.png)

**Comment by yaron2** [comment]
@63hogan hot-reloading of components is actually not an intended feature, and more of a bug at the moment.

The expected behavior is to perform a rolling update of your deployments after a change to a component. As such, we will be preventing the in-place updates in this release and introduce hot-reload (properly, with a feature flag) in the next release (1.8).

**Comment by msfussell** [comment]
#4433 PR closes this

**Comment by shubham1172** [comment]
@msfussell, #4433 closed #4432, which turns off component updates. However this bug still remains open. It should be fixed before we turn on components updates again. Can you please reopen the issue?
